src/dataset-builder/ConceptSearch.ts

```diff
@@ -0,0 +1,130 @@
+import { IconId } from '@terra-ui-packages/components';
+import _ from 'lodash/fp';
+import { Fragment, useEffect, useState } from 'react';
+import { div, h, h2 } from 'react-hyperscript-helpers';
+import { ActionBar } from 'src/components/ActionBar';
+import { Link, spinnerOverlay } from 'src/components/common';
+import { icon } from 'src/components/icons';
+import { SimpleTable } from 'src/components/table';
+import { StringInput } from 'src/data-catalog/create-dataset/CreateDatasetInputs';
+import { GetConceptsResponse } from 'src/dataset-builder/DatasetBuilderUtils';
+import { SnapshotBuilderConcept as Concept, SnapshotBuilderDomainOption } from 'src/libs/ajax/DataRepo';
+import { DataRepo } from 'src/libs/ajax/DataRepo';
+import { useLoadedData } from 'src/libs/ajax/loaded-data/useLoadedData';
+import colors from 'src/libs/colors';
+
+import { PAGE_PADDING_HEIGHT, PAGE_PADDING_WIDTH } from './constants';
+
+type ConceptSearchProps = {
+  readonly initialSearch: string;
+  readonly domainOption: SnapshotBuilderDomainOption;
+  readonly onCancel: () => void;
+  readonly onCommit: (selected: Concept[]) => void;
+  readonly onOpenHierarchy: (
+    domainOption: SnapshotBuilderDomainOption,
+    selected: Concept[],
+    searchText: string
+  ) => void;
+  readonly actionText: string;
+  readonly datasetId: string;
+  readonly initialCart: Concept[];
+};
+
+export const ConceptSearch = (props: ConceptSearchProps) => {
+  const { initialSearch, domainOption, onCancel, onCommit, onOpenHierarchy, actionText, datasetId, initialCart } =
+    props;
+  const [search, setSearch] = useState<string>(initialSearch);
+  const [cart, setCart] = useState<Concept[]>(initialCart);
+  const [concepts, searchConcepts] = useLoadedData<GetConceptsResponse>();
+  useEffect(() => {
+    void searchConcepts(() => {
+      return DataRepo().dataset(datasetId).searchConcepts(domainOption.root, search);
+    });
+  }, [search, datasetId, domainOption.root, searchConcepts]);
+  return h(Fragment, [
+    div({ style: { padding: `${PAGE_PADDING_HEIGHT}rem ${PAGE_PADDING_WIDTH}rem` } }, [
```

Comment:
I think this is the third component with exactly this same styling - do you think we should think about abstracting it to a display component?

Reply:
Good idea, it's used in four places now.

Reply:
Refactored into `BuilderPageHeader`